features/nanostack/FEATURE_NANOSTACK/mbed-mesh-api/source/thread_tasklet.c

```diff
@@ -228,67 +229,55 @@ void thread_tasklet_configure_and_connect_to_network(void)
         NET_6LOWPAN_THREAD);
 
     // Link configuration
-    memcpy(thread_tasklet_data_ptr->link_config.name, "Arm Powered Core", 16);
-
     thread_tasklet_data_ptr->link_config.panId = MBED_MESH_API_THREAD_CONFIG_PANID;
     TRACE_DETAIL("PANID %x", thread_tasklet_data_ptr->link_config.panId);
 
-    // channel
-    if (MBED_MESH_API_THREAD_CONFIG_CHANNEL > 27) {
-        tr_error("Bad channel %d", MBED_MESH_API_THREAD_CONFIG_CHANNEL);
-        return;
-    }
-
     thread_tasklet_data_ptr->link_config.rfChannel = MBED_MESH_API_THREAD_CONFIG_CHANNEL;
     thread_tasklet_data_ptr->channel_list.channel_page = (channel_page_e)MBED_MESH_API_THREAD_CONFIG_CHANNEL_PAGE;
     thread_tasklet_data_ptr->channel_list.channel_mask[0] = MBED_MESH_API_THREAD_CONFIG_CHANNEL_MASK;
     TRACE_DETAIL("channel: %d", thread_tasklet_data_ptr->link_config.rfChannel);
     TRACE_DETAIL("channel page: %d", thread_tasklet_data_ptr->channel_list.channel_page);
     TRACE_DETAIL("channel mask: %d", (int)thread_tasklet_data_ptr->channel_list.channel_mask[0]);
-
-
+    
     // Beacon data setting
     thread_tasklet_data_ptr->link_config.Protocol_id = 0x03;
     thread_tasklet_data_ptr->link_config.version = 1;
     memcpy(thread_tasklet_data_ptr->link_config.extended_random_mac, device_configuration.eui64, 8);
     thread_tasklet_data_ptr->link_config.extended_random_mac[0] |= 0x02;
 
+    // network name
+    MBED_ASSERT(strlen(MBED_MESH_API_THREAD_CONFIG_NETWORK_NAME) > 0 && strlen(MBED_MESH_API_THREAD_CONFIG_NETWORK_NAME) < 17);
+    memcpy(thread_tasklet_data_ptr->link_config.name, MBED_MESH_API_THREAD_CONFIG_NETWORK_NAME, 16);       
```

Comment:
Network Name 1
String in UTF-8 encoding matching the value field of Network Name TLV to provide 2 human-readable name for the network. Pad with null characters if the string is shorter 3 than 16 bytes.